Standard.AI.OpenAI.Tests.Unit/Services/Foundations/AIModels/AIModelServiceTests.Exceptions.RetrieveModelByName.cs

```diff
@@ -0,0 +1,298 @@
+﻿using System;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Moq;
+using RESTFulSense.Exceptions;
+using Standard.AI.OpenAI.Models.Services.Foundations.AIModels;
+using Standard.AI.OpenAI.Models.Services.Foundations.AIModels.Exceptions;
+using Xunit;
+
+namespace Standard.AI.OpenAI.Tests.Unit.Services.Foundations.AIModels
+{
+    public partial class AIModelServiceTests
+    {
+        [Fact]
+        public async Task ShouldThrowDependencyExceptionOnRetrieveAIModelByNameIfUrlNotFoundAsync()
+        {
+            // given
+            string someAIModelId = CreateRandomString();
+
+            var httpResponseUrlNotFoundException =
+                new HttpResponseUrlNotFoundException();
+
+            var invalidConfigurationAIModelException =
+                new InvalidConfigurationAIModelException(
+                    httpResponseUrlNotFoundException);
+
+            var expectedAIModelDependencyException =
+                new AIModelDependencyException(
+                    invalidConfigurationAIModelException);
+
+            this.openAIBrokerMock.Setup(broker =>
+                broker.GetAIModelByIdAsync(someAIModelId))
+                    .ThrowsAsync(httpResponseUrlNotFoundException);
+
+            // when
+            ValueTask<AIModel> retrieveAIModelByNameTask =
+               this.aiModelService.RetrieveAIModelByNameAsync(aiModelName: someAIModelId);
+
+            AIModelDependencyException
+                actualAIModelDependencyException =
+                    await Assert.ThrowsAsync<AIModelDependencyException>(
+                        retrieveAIModelByNameTask.AsTask);
+
+            // then
+            actualAIModelDependencyException.Should().BeEquivalentTo(
+                expectedAIModelDependencyException);
+
+            this.openAIBrokerMock.Verify(broker =>
+                broker.GetAIModelByIdAsync(someAIModelId),
+                    Times.Once);
+
+            this.openAIBrokerMock.VerifyNoOtherCalls();
+            this.dateTimeBrokerMock.VerifyNoOtherCalls();
+        }
+
+        [Fact]
+        public async Task ShouldThrowDependencyExceptionOnRetrieveAIModelByNameIfNotFoundAsync()
+        {
+            // given
+            string someAIModelId = CreateRandomString();
+
+            var httpResponseNotFoundException =
+                new HttpResponseNotFoundException();
+
+            var modelDoesNotExistException =
+                new ModelDoesNotExistException(
+                    httpResponseNotFoundException);
```

Comment:
```suggestion
            var notFoundAIModelException =
                new NotFoundAIModelException(
                    httpResponseNotFoundException);
```

I think this should be like this to align with our other services